mcs/class/System/System.Diagnostics/Win32EventLog.cs

```diff
@@ -736,7 +736,7 @@ private void NotifyEventThread (ManualResetEvent resetEvent)
 		{
```

Comment:
For me `if (resetEvent == null) return;` would make more sense

Reply:
It could become null inside the `while` loop. And checking with an if inside the while loop is racy as well which the null-conditional operator avoids.

Reply:
How can it become null inside the loop and not be a null before entering the loop?

Reply:
it is set to null in DisableNotification()

Reply:
This is a stack variable with DisableNotification having no influence to it

Reply:
right, for some reason I convinced myself that it's passed by `ref` but that isn't actually the case. Not sure how it's possible that resetEvent is null then?

Reply:
the race is here https://github.com/akoeplinger/mono/blob/6cc6ff04fa7fdf8cb2e11a15b268df26b67d7930/mcs/class/System/System.Diagnostics/Win32EventLog.cs#L729

it can called it with the `null` from there so the check could be placed there too but checking it inside the method looks more error proved for me